src/main/java/javaslang/collection/List.java

```diff
@@ -1202,7 +1202,7 @@ default T peek() {
     }
 
     @Override
-    default List<T> unit(Iterable<? extends T> iterable) {
+    default List<T> cons(Iterable<? extends T> iterable) {
```

Comment:
@mperry What do you think of having a `Monad` / `FilterMonadic` interface with methods (in conjunction with the `Kind`for semi-higher order kinds`):

* `flatMapM`
* `flattenM`

Have these methods a different meaning, too, in other languages?

I can't use the original names `flatMap` and `flatten` because they have a different, relaxed signature (with Iterable instead of Monad/FilterMonadic for  Traversables).

Thanks in advance!